library/alloc/src/string.rs

```diff
@@ -1489,10 +1489,11 @@ impl String {
         Some(ch)
     }
 
-    /// Removes a [`char`] from this `String` at a byte position and returns it.
+    /// Removes a [`char`] from this `String` at byte position `idx` and returns it.
     ///
-    /// This is an *O*(*n*) operation, as it requires copying every element in the
-    /// buffer.
+    /// Copies all bytes after the removed char to new positions.
+    ///
+    /// NB Calling this in a loop can result in quadratic behavior.
```

Comment:
Could you replace "NB" with "Note that"? We've had some people be confused with "NB" before, and it reads a bit unusual in prose ("N.B." may be better, but not much reason not to just spell it out)